js/DataSourceSorterComposite.js

```diff
@@ -49,22 +49,42 @@ var DataSourceSorterComposite = DataSourceIndexed.extend('DataSourceSorterCompos
         this.sorts.push({ columnIndex: columnIndex, direction: direction });
     },
 
-    set: function(sorts) {
-        this.sorts = sorts || [];
+    /**
+     *
+     * @memberOf DataSourceSorterComposite#
+     * @param {sorterFunction} [sorter] - If undefined, deletes sorter.
+     */
+    set: function(sorter) {
+        if (sorter) {
+            /**
+             * @implements sorterInterfacei
+             * @memberOf DataSourceSorterComposite#
+             */
+            this.sorter = sorter;
+        } else {
+            delete this.sorter;
+        }
+    },
+
+    get: function() {
+        return this.sorter;
     },
 
     /**
      * @memberOf DataSourceSorterComposite#
      */
     apply: function() {
         var each = this.dataSource;
-
-        this.sorts.forEach(function(sortSpec) {
-            each = new DataSourceSorter(each);
-            each.sortOn(sortSpec.columnIndex, sortSpec.direction);
-        });
-
-        this.last = each;
+        // get list of sorts from either API or use existing
+        this.sorts = (this.sorter && this.sorter.prop('sorts')) || this.sorts;
+
+        if (this.sorts) {
+            this.sorts.forEach(function(sortSpec) {
+                each = new DataSourceSorter(each);
+                each.sortOn(sortSpec.columnIndex, sortSpec.direction, sortSpec.type);
```

Comment:
For type we should be using `schema[sortSpec.columnIndex].type`.

Passing `type` in the `sortSpec` is OK I guess but shouldn't be necessary. So if it is passed, we need to decide which gets priority. I'm thinking the passed value would override the schema value.